### PR TITLE
feat: add kubevela cli spec

### PR DESCRIPTION
**Pull request by basuotian**
Signed-off-by: tianshuai 

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
kubevela is A Highly Extensible Platform Engine based on Kubernetes and Open Application Model.
Feature to support vela cli

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**
support vela (cli for kubevela) autocomplete

**Additional info:**
vela cli info: https://kubevela.net/docs/cli/vela

### CODE REVIEW

File: src/vela.ts [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+      description:
+        "Show the reference doc for component, trait or workflow types",
+      args: {
+        isVariadic: true,
+        template: ["history"],
+      },
+      options: [
+        {
+          name: ["-e", "--env"],
+          description: "Specify environment name for application",
```
Comment (fedeci) [inline src/vela.ts:17]: This accepts an argument

File: src/vela.ts [hunk 2]
```diff
@@ -0,0 +1,773 @@
...
+          description: "If present, the namespace scope for this CLI request",
+        },
+        {
+          name: ["--publish-version", "-v"],
+          description: "The publish version for deploying application",
+        },
+        {
+          name: ["--revision", "-r"],
+          description:
+            "The revision to use for deploying the application, if empty, the current application configuration will be used",
```
Comment (fedeci) [inline src/vela.ts:120]: All these accept an argument

File: src/vela.ts [hunk 3]
```diff
@@ -0,0 +1,773 @@
...
+        {
+          name: "detach",
+          description: "Detach managed cluster",
+          options: [
+            {
+              name: ["--kubeconfig-path", "-p"],
+              description:
+                "Specify the kubeconfig path of managed cluster. If you use ocm to manage your cluster, you must set the kubeconfig-path",
+            },
+          ],
```
Comment (fedeci) [inline src/vela.ts:141]: An arg is missing here too

**Review by fedeci — CHANGES_REQUESTED** [review]
Hey @basuotian, thanks for this spec! I am seeing a lot of args missing throughout the file, can you add those, please? I have not marked all the missing ones but only gave you a few examples. For the rest the spec looks great!

**Comment by basuotian** [comment]
sorry for missing the args, the latest commit has resolved, PTAL, thx

**Review by fedeci — APPROVED** [review]
Thanks a lot for the changes!

**Comment by SeparateRecords** [comment]
Thanks for the contribution @basuotian! Looks like a lot of work to have written this by hand. Awesome work.

**Pull request merged**